src/networkingSockets.cpp

```diff
@@ -0,0 +1,460 @@
+        return 0;
+    }
+    luasteam::pushuint64(L, pIdentity->GetSteamID().ConvertToUint64());
+    return 1;
+}
+
+// HSteamNetPollGroup CreatePollGroup()
+EXTERN int luasteam_createPollGroup(lua_State *L) {
+    HSteamNetPollGroup pollGroup = steamNetworkingSocketsLib()->CreatePollGroup();
+    lua_pushinteger(L, pollGroup);
+    return 1;
+}
+
+// DestroyPollGroup( HSteamNetPollGroup hPollGroup )
+EXTERN int luasteam_destroyPollGroup(lua_State *L) {
+    HSteamNetPollGroup hPollGroup = luaL_checkinteger(L, 1);
+    bool result = steamNetworkingSocketsLib()->DestroyPollGroup(hPollGroup);
+    lua_pushboolean(L, result);
+    return 1;
+}
+
+// SetConnectionPollGroup( HSteamNetConnection hConn, HSteamNetPollGroup hPollGroup )
+EXTERN int luasteam_setConnectionPollGroup(lua_State *L) {
+    HSteamNetConnection hConn = luaL_checkinteger(L, 1);
+    HSteamNetPollGroup hPollGroup = luaL_checkinteger(L, 2);
+    bool result = steamNetworkingSocketsLib()->SetConnectionPollGroup(hConn, hPollGroup);
+    lua_pushboolean(L, result);
+    return 1;
+}
+
+// int ReceiveMessagesOnPollGroup( HSteamNetPollGroup hPollGroup, SteamNetworkingMessage_t **ppOutMessages, int nMaxMessages )
+EXTERN int luasteam_receiveMessagesOnPollGroup(lua_State *L) {
+    HSteamNetPollGroup hPollGroup = luaL_checkinteger(L, 1);
+
+    SteamNetworkingMessage_t *msgs[32];
+    int numMessages = steamNetworkingSocketsLib()->ReceiveMessagesOnPollGroup(hPollGroup, msgs, 32);
+
+    lua_pushinteger(L, numMessages);
+    if (numMessages >= 0) {
+        lua_createtable(L, 0, numMessages);
+        for (int i = 0; i < numMessages; i++) {
+            SteamNetworkingMessage_t *message = msgs[i];
+            uint32 messageSize = message->GetSize();
+
+            lua_createtable(L, 0, 2);
+            if (messageSize > 0) {
+                lua_pushlstring(L, (char *)message->GetData(), messageSize);
+            } else {
+                lua_pushstring(L, "");
+            }
+            lua_setfield(L, -2, "msg");
+            lua_pushinteger(L, message->m_conn);
+            lua_setfield(L, -2, "conn");
+            lua_rawseti(L, -2, i + 1);
+            message->Release();
+        }
+        return 2;
+    }
+    return 1;
+}
+
+// FlushMessagesOnConnection( HSteamNetConnection hConn )
+EXTERN int luasteam_flushMessagesOnConnection(lua_State *L) {
+    HSteamNetConnection hConn = luaL_checkinteger(L, 1);
+    EResult result = steamNetworkingSocketsLib()->FlushMessagesOnConnection(hConn);
+    lua_pushstring(L, steam_result_code[result]);
+    return 1;
+}
+
+// void SendMessages( int nMessages, SteamNetworkingMessage_t *const *pMessages, int64 *pOutMessageNumberOrResult )
+EXTERN int luasteam_sendMessages(lua_State *L) {
+    int nMessages = luaL_checkinteger(L, 1);
+    int64 *pOutMessageNumberOrResult = new int64[nMessages];
+    SteamNetworkingMessage_t **pMessages = new SteamNetworkingMessage_t *[nMessages];
```

Comment:
IIUC from the API, steam takes ownership of these arrays and will free them, correct? It's a bit confusing when we see no free's anywhere. If so, it might be nice to at least add a comment explaining that.